lib/compiler/passes/generate-bytecode.js

```diff
@@ -197,18 +196,42 @@ const visitor = require("../visitor");
 //
 //        silentFails--;
 function generateBytecode(ast) {
-  const consts = [];
+  const literals = [];
+  const classes = [];
+  const expectations = [];
+  const functions = [];
 
-  function addConst(value) {
-    const index = consts.indexOf(value);
+  function addLiteralConst(value) {
```

Comment:
I sure wish we could use a `Set` for this.

Reply:
`Set` does not generally offer the performance bonus you'd expect, because it isn't actually a `Set`, but rather an array which features a `Set` API

Just remember: everything about Javascript is deeply broken

v8 has lots of performance magic around this, but fundamentally, we can't do tuning like that in the real world without beefy benchmarks

Reply:
I think, using `Set` here is impossible, because, while `Set` preserve insertion order, it have no way to get an index of an already contained object.

We could use `Map<object, number>` here, to map from object to it index, but I think that will be a premature optimization. In the end, number of objects, contained in that collections, hardly ever exceed even 1000. Using an ordinally array on such counts seems reasonable.

Reply:
Don't worry about making a change here.